### ISSUE TITLE
Replace image of video player with 'No preview available' - aka ICONS everywhere!

### ISSUE THREAD

**Issue by orangewolf** [notch8/atla_digital_library#202]
# Updated current behavior - what is left to do on this ticket
- default icons for videos and audio file works are appearing as the thumbnail, even when they have a thumbnail specified. 
- specifed thumbnails should show when they are specified as opposed to the default ones

# Related
This ticket is a duplicate: https://github.com/scientist-softserv/atla_digitial_library/issues/17
Completing the work on this ticket should also close #17.

# Acceptance Criteria
- [x] Newly imported works with the type "Moving Image" with an image url in the `remote_files` column of a bulkrax  import will have that image as a thumbnail.
- [x]  Newly imported works with the type "Sound" with an image url in the `remote_files` column of a bulkrax  import will have that image as a thumbnail.
- [x] Newly imported works with the type "Moving Image" with an image url in the `remote_files` column of a bulkrax import will have the default video image as a thumbnail.
- [x]  Newly imported works with the type "Sound" without an image url in the `remote_files` column of a bulkrax  import will have the default audio image as a thumbnail.

# Testing instructions
1. Download the following test csv as a csv: https://docs.google.com/spreadsheets/d/1D2HqvPJhDKorqtK8pz8UQzPc-fmWlcnSeeTPKXmD_DQ/edit#gid=0
2. navigate to /dashboard
3. click "importers" on the left panel
4.  Upload the csv to bulkrax - choose the csv parser
5. Make sure the default sound and default video icons show up for the sound and moving image works with NO thumbnail
6. Check that adventure time thumbnails show up for the sound and moving image that are titled "with thumbnail"
7. The work titled "text - no thumbnail" should have the default box thumbnail, and the text & article "with thumbnail" should both have adventure time thumbnails

# Old behavior (this has since been fixed)

<img src="https://user-images.githubusercontent.com/1054448/202233232-f8656b92-7e99-4b3e-81eb-8241b122a0d0.png" width=500 />

There is an image of a video player, which is unnecessary because users are not able to click the on the player to view the content. 

# Old Expected Behavior (this has been replaced by the use of icons)
The text 'No preview available' is preferred like how it displays for audio: https://dl.atla.com/concern/works/000005133?locale=en

<img src="https://user-images.githubusercontent.com/1054448/202233467-814bd1ae-b581-4470-baa4-37aa826b59fe.png" width=500 />

**Comment by orangewolf** [comment]
Related MRs:

- [hide the video player and any matched audio](https://gitlab.com/notch8/atla_digital_library/-/merge_requests/244)
!244

Christy Karpinski - February 2020
---
Quesiton on this - would it be possible to instead put an icon as thumbnail instead of "no preview available"?
Just learning from the UX sessions that the "no preview available" is confusing to users - they end up not knowing what to do to see the item and click on nothing or eventually find their way down to the URL field
If it is related and possible to include here - any works that use no preview available, if that could instead show an icon would be great. Ideally it would be content type related but the generic work icon would be better than the text.

**Comment by orangewolf** [comment]
Rob Kaufman - March 2020
---


![Image](https://user-images.githubusercontent.com/1054448/202234973-5b17d4bd-4b73-43bd-abdc-c7ab0454fa47.png)

**Comment by orangewolf** [comment]
Christy Karpinski - March 2020
---


![Image](https://user-images.githubusercontent.com/1054448/202235273-e766b447-6d9b-43a7-bd0b-ccc2eb9d1220.png)



![Image](https://user-images.githubusercontent.com/1054448/202235370-fa049d32-fd37-4e3a-bfe0-49f51646aee5.png)

I presume we could swap these for something else later if we wanted to, is that true?

**Comment by orangewolf** [comment]
Christy Karpinski - March 2020
---
Here is what I had put in to the other ticket

WHy do some audio/video collections show this icon for the work page

![Image](https://user-images.githubusercontent.com/1054448/202235831-9a599d57-3567-4c0f-b112-c9edd94081d3.png)

And others (i think all Princeton) show "no preview available"? like this


![Image](https://user-images.githubusercontent.com/1054448/202236076-18564e05-05c1-4110-80c6-dbddc6c5ef53.png)


Can we get the icon image that shows here:


![Image](https://user-images.githubusercontent.com/1054448/202236234-973cfb51-4585-4b17-894f-a918998d700b.png)


To be the same as what shows when you click that item and view its metadata page here


![Image](https://user-images.githubusercontent.com/1054448/202236456-1429271c-f71c-4f70-8b29-547842cdb29b.png)


WE ALSO WANT to change that block file - it is the work.png file, we are looking at replacing it with something better. BUT possibly if it would show the blue outlined document file instead of the purple blocks that would be the answer

* if the only time this default icon shows is when the works are audio or video (because all others will have thumbnails as we require) we may want to make this into a media file something like this (updated this file to solid grey with fully transparent background on 7/16/2020)



![Image](https://user-images.githubusercontent.com/1054448/202237075-66757faa-bf72-4b97-bf55-725e66cfedf0.png)



* I think ideally we would like the icon to show audio for audio, video for video - which is probably a good amount of work?

**Comment by orangewolf** [comment]
Crystal - April 2020
---
This will cover the new works that are imported, but will not fix previous works. Rob will fix previous versions with #277.This will cover the new works that are imported, but will not fix previous works.  Rob will fix previous versions with [#277 (closed)](https://github.com/notch8/atla_digital_library/-/issues/277).

**Comment by orangewolf** [comment]
Christy Karprinski - May 2020
---
1. right now there is an orange button that says "download image" but it is a video https://dl.atla.com/concern/works/5999n920k?locale=en

Can we get this to maybe just say "Download file"



![Image](https://user-images.githubusercontent.com/1054448/202238043-a1fe6959-e699-4c55-a069-5d87f390db06.png)


1. Also there are two CDRI collections that are HTML, can those get the HTML icon that i added to the collection itself (attached)?

https://dl.atla.com/collections/selected-sermons-preached-at-princeton-theological-seminary-during-the-nineteenth-century?locale=en

https://dl.atla.com/collections/christian-missions-and-social-progress?locale=en


![Image](https://user-images.githubusercontent.com/1054448/202238249-f5d6e653-69d5-4c6e-b350-16eed7244da9.png)

1. These icons show on the item show page but still dont correlate the collection list view https://dl.atla.com/collections/worship-video-project - is that going to show them at some point too?

**Comment by orangewolf** [comment]
Christy Karpinski - July 2020
---
Page mock up I made for a presentation - adding here in case it is helpful. This would be the icon to use for audio and video works if we need to use the same one for both. AND "Download the file" as button text instead of download image


![Image](https://user-images.githubusercontent.com/1054448/202238745-0d5d6d4e-6779-45b6-8873-324415ef65db.png)

**Comment by orangewolf** [comment]
Crystal - August 2020
Will create a different ticket for Hyrax upgrade.

**Comment by orangewolf** [comment]
Maria Dougherty - October 2020
---
**Pass** - an icon appears if there is no media available.

<details>
<summary></summary>
</details>


![Image](https://user-images.githubusercontent.com/1054448/202247821-53d3d51e-1bbe-4e5f-bf66-2e64068737af.png)


![Image](https://user-images.githubusercontent.com/1054448/202248293-bc0a8aaf-3dab-43ef-b23f-19cb623c121e.png)

**Comment by orangewolf** [comment]
Christy Karpinski - October 2020
---
@Rob so I thought this was still lingering because

1 either you had found a way to put in the audio/video icon when the file type was audio/video - beyond CDRI Or

2 we were going to swap the default icon to be the audio/video icon (would only be a problem for the few html file types we have in CDRI and I could probably fix those manuall)

3 Part of the "icons everywhere" was that there is a different icon showing (purple block) in list view vs on the item record - thought we were going to get those to match the same way a thumbnail images matches

4 ALso we need the text on the button for download to say download file instead of download image since it is almost never an image file that is having the no icon.

My guess is we cant do 1 or 3 easily. 2 I think I could do myself 3 Maybe i can find where that text is and do myself

Ill look into those tomorrow and report back what I find[@Rob](https://github.com/Rob) so I thought this was still lingering because
1 either you had found a way to put in the audio/video icon when the file type was audio/video - beyond CDRI
Or
2 we were going to swap the default icon to be the audio/video icon (would only be a problem for the few html file types we have in CDRI and I could probably fix those manuall)
3 Part of the "icons everywhere" was that there is a different icon showing (purple block) in list view vs on the item record - thought we were going to get those to match the same way a thumbnail images matches
4 ALso we need the text on the button for download to say download file instead of download image since it is almost never an image file that is having the no icon.
My guess is we cant do 1 or 3 easily.
2 I think I could do myself
3 Maybe i can find where that text is and do myself
Ill look into those tomorrow and report back what I find

**Comment by orangewolf** [comment]
Rob Kaufman - April 2022
---
ok, I think we are down to 2 lingering things here:

1. we use the blue document / purple boxes for any item where there is no file attached. Its clear to me that you want the icon to be the same in both the thumbnail AND the show page and you want the puple box to go away, but it is not clear if that thumbnail should be the blue page icon or the A/V icon attached here.
1. the "2012 MidWinters" collection specifically does not have any file attachments which is why we don't know those are A/V files and they don't get icons. Depending on how (1) shakes out, this may become a moot point.

again, to be clear, I think all the other parts of this ticket are resolved and we're down to just the two part question about works which do not have files attached.

**Comment by orangewolf** [comment]
Christy Karpinski - April 2022
---
@orangewolf

1. I think the AV icon should only be used if the items are A/V. If that can happen, then I think the only time we would see the blue doc or purple box is when something is missing/not correct - does that sound right?

If that is the case then it seems it would be useful to keep it as purple box/blue icon

1. is it possible to use the "MP4" in format (digital) as a clue to attach the AV icon? If no file is attached, look at the format (digital)?[@orangewolf](https://github.com/orangewolf)

I think the AV icon should only be used if the items are A/V. If that can happen, then I think the only time we would see the blue doc or purple box is when something is missing/not correct - does that sound right?

If that is the case then it seems it would be useful to keep it as purple box/blue icon

is it possible to use the "MP4" in format (digital) as a clue to attach the AV icon? If no file is attached, look at the format (digital)?

**Comment by orangewolf** [comment]
Christy Karpinski - August 2022
---
This princeton collection is audio and should have the AV icons https://dl.atla.com/collections/earl-palmer-collection?locale=en

It does not have MP4 in the format digital, if no file is attached maybe it could be - check for file type, then check for format (digital), then check type if it is Sound or Moving Image?

**Comment by orangewolf** [comment]
Christy Karpinski - August 2022
---
@orangewolf it seems that there are instances where there were existing thumbnail images that were replaced by the av icon. This is not what we wanted. The intent was if the work had a thumbnail it would be skipped.

Here is an example https://dl.atla.com/concern/works/jm215126b?locale=en

Go to the bottom of this search result page to the audio icons - all those have thumbnails and should not have the audio icon https://dl.atla.com/collections/david-l-smiley-papers?locale=en&page=3&per_page=100&sort=score+desc%2C+system_create_dtsi+desc

**Comment by orangewolf** [comment]
Christy Karpinkski - August 2022
---
Same with this collection - they gave us a file to use for the thumbnail, it got replacedhttps://dl.atla.com/collections/sermons-of-the-reverend-dr-william-augustus-jones-jr

We want the file that we uploaded with the metadata to be used for these

**Comment by ckarpinski** [comment]
This collection also will need fixing - I imported this with a thumbnail image to be applied to all works, on the colleciton page that lists the works it has been replaced with the AV icon https://dl.atla.com/collections/a-life-apart-hasidism-in-america

**Comment by summer-cook** [comment]
notes:
- if there is a thumbnail, it should always be used. check for thumbnail needs to happen first, and then if there is NO thumbnail, work-type icon will be used as backup.
- issue https://github.com/scientist-softserv/atla_digitial_library/issues/17 is a potential duplicate of this issue/its likely that fixing the issue listed above will also fix 17.

**Comment by summer-cook** [comment]
# Softserv QA: Passed



![Image](https://user-images.githubusercontent.com/73361970/232848481-a557df36-1340-4072-b35d-8f3120517969.png)


Upon import, works that include thumbnails show up with thumbnails, otherwise they show the default thumbnail based on their worktype.

**Comment by ckarpinski** [comment]
I tried re importing the Earl Palmer collection to see if I could get the icons for it correct. It is an OAI collection where some items have thumbnails and the a/v do not.  It looks to me like it removed the default icon so now there is no thumbnail on any of the av
https://dl.atla.com/collections/earl-palmer-collection

This issue was created because of this collection and others like it from the same contributor (https://dl.atla.com/collections/princeton-theological-seminary-wright-library) - we want those works with no thumbnail to have the default video or audio one. (The CDRI collections were the other use case for this https://dl.atla.com/collections/cooperative-digital-resources-initiative)

here is the importer https://dl.atla.com/importers/92?locale=en

**Comment by ckarpinski** [comment]
This did work when i did a csv import with thumbnail file

**Comment by ckarpinski** [comment]
I am making new tickets for the issues that have come up as I rerun collections to fix their av thumbnails
